hw/ip/clkmgr/doc/_index.md

```diff
@@ -238,8 +238,17 @@ Software sets both an expected maximum and minimum for each measured clock.
 Clock manager then counts the number of relevant root clock cycles in each always-on clock period.
 If the resulting count differs from the programmed thresholds, a recoverable error is registered.
 
-Additionally, clock manager uses a similar time-out mechanism to see if any of the root clocks have stopped toggling altogether.
-This is done by creating an artificial handshake between the two domains that must complete within a certain amount of time based on known clock ratios.
+Since the counts are measured against a single cycle of always on clock, the minimal error that can be detected is dependent on the clock ratio between the measured clock and 1 cycle of of the always on clock.
+Assume a 24MHz clock and an always-on clock of 200KHz.
+The minimal error detection is then 200KHz / 24MHz, or approximately 0.83%.
+
+This means if the clock's actual value is between 23.8MHz and 24.2MHz, this deviation will not be detected.
+Conversely, if the clock's natural operation has an error range wider than this resoltion, the min / max counts must be adjusted to account for this error.
+
+Additionally, clock manager uses a similar time-out mechanism to see if any of the root clocks have stopped toggling for an extended period of time.
+This is done by creating an artificial handshake between the two root clock domain and the always on clock domain that must complete within a certain amount of time based on known clock ratios.
+Based on the nature of the handshake and the margin window, the minimal timeout detection window is approximately 2-4 always on clock cycles.
+If the root clock domain stops and resumes in significantly less time than this window, the time-out may not be detected.
 
```

Comment:
this is actually a bigger point i need to state in the spec.
We can't actually turn on these checks when we are going to step-down clocks.

The timing of those step changes right now are not synchronized to the measurements, which means they could come at point where the count is already way too big. Instead of fixing this now though (probably too late in the game..) i'm going to add a clarifying statement that the frequency measurements should not be used together with the clock switches. 

Since the clock switches are really meant for debugging only, this is probably an acceptable compromise for now.